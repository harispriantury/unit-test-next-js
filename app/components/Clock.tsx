"use client"
import React, { useEffect, useState } from 'react'

const Clock = () => {
    const [time, setTime] = useState<string>('');

    setTimeout(() => {
        const newTime = new Date();
        setTime(newTime.toLocaleTimeString())
    }, 1000)
    return (
        <div>{time}</div>
    )
}

export default Clock