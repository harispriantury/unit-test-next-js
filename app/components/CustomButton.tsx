import React from 'react'

interface CustomButtonProps {
    text: string
    style: string
}

export const CustomButton = ({ text, style }: CustomButtonProps) => {
    return (
        <button className={`px-4 py-3 ${style} shadow-lg`}>
            {text}
        </button >
    )
}
