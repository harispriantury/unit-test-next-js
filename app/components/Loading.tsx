import React, { FC } from 'react'

interface ILoading {
    isActive: boolean
}

const Loading: FC<ILoading> = ({ isActive }) => {
    return (
        <div>


        </div>)
}

export default Loading