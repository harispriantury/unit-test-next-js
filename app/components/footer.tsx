import React from 'react'

const Footer = () => {
    return (
        <footer className='absolute right-0 w-full bg-greenDark text-center text-white p-5'>
            <h1>Copyright © 2023. Made by moslem from Indonesia with ❤️</h1>
        </footer>
    )
}

export default Footer