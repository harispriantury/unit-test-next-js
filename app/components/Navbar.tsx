"use client"
import React from 'react'
import Image from 'next/image'
import logo from '@/public/islam2.png'
import Link from 'next/link'
import { CustomButton } from './CustomButton'
import { motion } from 'framer-motion'
import { usePathname } from 'next/navigation'

interface INavList {
    name: string
    url: string
}

const navList: INavList[] = [
    {
        name: 'Halaman Utama',
        url: '/'
    },
    {
        name: `Al-Qur'an`,
        url: '/quran'
    },
    {
        name: 'Jadwal Sholat',
        url: '/jadwal'
    }
]

export const Navbar = () => {
    const pathname = usePathname();
    return (
        <nav className='z-50 sticky w-full top-0 right-0 left-0 grid grid-cols-8 bg-green items-center mx-auto text-white border-b-2 border-greenDark'>
            <div className='flex justify-between items-center p-4 col-span-2 font-semibold'>
                <Link href={'/'} className='flex items-center gap-2 text-2xl'>
                    <Image src={logo} alt={''} width={50} />
                    <span>Aplikasi Al-Qur`an</span>
                </Link>
            </div>
            <ul id='navlist' className='col-span-4 flex justify-center gap-5'>
                {
                    navList.map((item: INavList) => {
                        return (
                            <li key={item.name} className='text-xl'>
                                <Link href={item.url} className={`relative ${pathname === item.url ? 'opacity-100' : 'opacity-80'}`}>
                                    {item.name}
                                    {
                                        pathname === item.url && (
                                            <motion.span
                                                layoutId="underline"
                                                className="absolute block h-[0.5px] w-full bg-white"
                                            />
                                        )
                                    }
                                </Link>
                            </li>

                        )
                    })
                }

            </ul>
            <div className='col-span-2 flex justify-end'>
                <CustomButton text='login' style='' />
            </div>
        </nav>
    )
}
