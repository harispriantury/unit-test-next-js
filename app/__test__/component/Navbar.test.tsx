import { render, screen } from "@testing-library/react";
import { Navbar } from "../../components/Navbar";
import "@testing-library/jest-dom";

it("renders Navbar Component", () => {
  const page = render(<Navbar />);
  expect(page).toMatchSnapshot();
});
