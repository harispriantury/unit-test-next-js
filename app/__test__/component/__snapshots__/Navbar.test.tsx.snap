// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Navbar Component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="z-50 sticky w-full top-0 right-0 left-0 grid grid-cols-8 bg-green items-center text-white border-b-2 border-greenDark py-10 lg:py-2"
      >
        <div
          class="flex justify-between items-center p-4 col-span-2 font-semibold"
        >
          <a
            class="flex items-center gap-2 md:text-lg xl:text-xl w-full"
            href="/"
          >
            <img
              alt=""
              data-nimg="1"
              decoding="async"
              height="50"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=128&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=64&q=75 1x, /_next/image?url=%2Fimg.jpg&w=128&q=75 2x"
              style="color: transparent;"
              width="50"
            />
            <span
              class="whitespace-nowrap"
            >
              Aplikasi Al-Qur\`an
            </span>
          </a>
        </div>
        <ul
          class="col-span-4 flex justify-center gap-5 max-lg:hidden"
          id="navlist"
        >
          <li
            class="text-xl"
          >
            <a
              class="relative opacity-80"
              href="/"
            >
              Halaman Utama
            </a>
          </li>
          <li
            class="text-xl"
          >
            <a
              class="relative opacity-80"
              href="/quran"
            >
              Al-Qur'an
            </a>
          </li>
          <li
            class="text-xl"
          >
            <a
              class="relative opacity-80"
              href="/jadwal"
            >
              Jadwal Sholat
            </a>
          </li>
        </ul>
        <div
          class="lg:hidden col-span-2 col-start-6 text-4xl flex justify-end text-white"
          id="menu"
        >
          <div>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M88 152h336M88 256h336M88 360h336"
                fill="none"
                stroke-linecap="round"
                stroke-miterlimit="10"
                stroke-width="48"
              />
            </svg>
          </div>
        </div>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="z-50 sticky w-full top-0 right-0 left-0 grid grid-cols-8 bg-green items-center text-white border-b-2 border-greenDark py-10 lg:py-2"
    >
      <div
        class="flex justify-between items-center p-4 col-span-2 font-semibold"
      >
        <a
          class="flex items-center gap-2 md:text-lg xl:text-xl w-full"
          href="/"
        >
          <img
            alt=""
            data-nimg="1"
            decoding="async"
            height="50"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=128&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=64&q=75 1x, /_next/image?url=%2Fimg.jpg&w=128&q=75 2x"
            style="color: transparent;"
            width="50"
          />
          <span
            class="whitespace-nowrap"
          >
            Aplikasi Al-Qur\`an
          </span>
        </a>
      </div>
      <ul
        class="col-span-4 flex justify-center gap-5 max-lg:hidden"
        id="navlist"
      >
        <li
          class="text-xl"
        >
          <a
            class="relative opacity-80"
            href="/"
          >
            Halaman Utama
          </a>
        </li>
        <li
          class="text-xl"
        >
          <a
            class="relative opacity-80"
            href="/quran"
          >
            Al-Qur'an
          </a>
        </li>
        <li
          class="text-xl"
        >
          <a
            class="relative opacity-80"
            href="/jadwal"
          >
            Jadwal Sholat
          </a>
        </li>
      </ul>
      <div
        class="lg:hidden col-span-2 col-start-6 text-4xl flex justify-end text-white"
        id="menu"
      >
        <div>
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M88 152h336M88 256h336M88 360h336"
              fill="none"
              stroke-linecap="round"
              stroke-miterlimit="10"
              stroke-width="48"
            />
          </svg>
        </div>
      </div>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
